Data/ByteString.hs

```diff
@@ -1078,6 +1080,20 @@ index ps n
     | otherwise      = ps `unsafeIndex` n
 {-# INLINE index #-}
 
+-- | 'ByteString' index that returns 'Nothing' if:
+--      @n < 0@
+--      @n > length byteString@
```

Comment:
That doesn't look like valid haddock markup; have you checked this renders as you'd expect it to?

Moreover, a `@since 0.10.10.0` annotation is desirable.